proto/chainlink/v1beta/tx.proto

```diff
@@ -2,18 +2,52 @@ syntax = "proto3";
 package chainlink.v1beta;
 
 import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
 
 option go_package = "github.com/ChainSafe/chainlink-cosmos/x/chainlink/types";
 
-// Msg defines the Msg service.
+// Msg defines the Msg gPRC service.
 service Msg {
-    // this line is used by starport scaffolding # proto/tx/rpc
+  rpc SubmitFeedData(MsgFeedData) returns (MsgFeedDataResponse) {
+    option (google.api.http).put = "/chainlink/feed/data";
+  }
 }
 
-message MsgFeedData {
+// MsgFeed is the type defined for each feed
+message MsgFeed {
+  // FeedId is the unique identifier of the feed
   string feedId = 1;
+  // FeedOwner is the owner of the feed
   bytes feedOwner = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
-  OCRAbiEncoded feedData = 3;
+  // DataProviders is the whitelist of data providers of the feed
+  repeated DataProvider dataProviders = 3;
+  // TODO: add feed parameters, reward schema etc.
+}
+
+// DataProvider is the type defined for feed data provider
+message DataProvider {
+  bytes address = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+  bytes pubKey = 2;
+}
+
+// MsgFeedData is the type defined for the data of the feed
+// It could be an OCR report feed, or any general feed data in the future
+message MsgFeedData {
+  // FeedId is the unique identifier of the feed
+  string feedId = 1;
+  // Submitter is the leader of the current round who is calling the module to submit feed data
+  bytes submitter = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+  // FeedData is the report for price feed, could be used for general feed data
+  // From proposal: The reports generated by Chainlink nodes off-chain using the OCR protocol are ABI encoded byte arrays
+  // which means each report is a byte array which could be deserialized to OCRAbiEncoded type
+  bytes feedData = 3;
```

Comment:
i think this should be
`repeated RoundData = 3;`

Reply:
or i think youre using feedData as `bytes` to accept dummy data... right?
if so, let's just add a TODO comment of the correct implementation

Reply:
nvm, this `MsgFeedData` is going inside `OCRFeedDataInStore`...
so `feedData` here should be renamed to something like `roundData` or something... what do you think?

Reply:
pls look at the comment above `feedData`, the proposal mentioned clearly, the each report data is a `byte array` when it gets submitted to the module, we need to deserialize it(using Go lib we will build later) based on the feed type, currently we only force on price feed, but this `feedData` could be used for any type or report in the future.

Reply:
ok i think i got confused because its the same var name in 
```
message OCRFeedDataInStore {
  uint64 RoundId = 3;
  MsgFeedData feedData = 1;
  OCRAbiEncoded deserializedOCRReport = 2;
}
```

are these supposed to be the same or should we label them more distinctively?

Reply:
so the idea is that the `feedData` in `MsgFeedData` will always be `bytes`, but it could be any feed data including price feed after deserialized, in our case, it's `OCRAbiEncoded`. `OCRFeedDataInStore` is the type that we need to marshel it and persist into store. As you can see, `OCRFeedDataInStore` has a concrete type specific for OCR report, which I think should be a interface type, so that any feed data(not limited to OCR) can be marshelled and stored.

Reply:
ok, so you are saying we need the same variable names because they may refer to the same thing but sometimes it might be within an OCRAbiEncoded object...?

Reply:
yeah kinda, we can leave them like this for now, you will have a better picture when we get into the next milestone